### PR TITLE
Update standard to remove eslint-plugin-node

### PR DESCRIPTION
**Pull request by Qard**
### What does this PR do?

Update to latest version of `eslint-config-standard` so we can eliminate `eslint-plugin-node`.

### Motivation

`eslint-plugin-node` triggers vuln scanner warnings so we should get it out of our dependency tree.

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by tlhunter — APPROVED** [review]

**Comment by rochdev** [comment]
Just a thought, but wouldn't it be a much smaller change to adjust the rules to be as they were before instead of changing 306 files? Not against the change per se, but it's very likely to cause conflicts with every open branch, and the goal of the PR is to fix a vulnerability and not to change our rules.

**Comment by Qard** [comment]
The whole _point_ of using standard is to not debate style issues and just adopt what it says, so if we go and carve out a bunch of exceptions that seems like it's opening up debate about which rules we should or should not follow. Personally I felt it was better to just make the changes to adapt, though I _did_ go for just using a bunch of per-line rule exceptions in a bunch of places just to keep the delta relatively minimal.

If we would prefer to just use standard as a base and build our own ruleset on top of that then we should probably have a discussion here about which of the new rules should be kept and which should be changed. Thoughts?

**Comment by Qard** [comment]
Apparently I completely missed that #4032 already existed. We could just land _that_ if you prefer how it did things. 🤔

**Comment by rochdev** [comment]
> We could just land that if you prefer how it did things.

I don't have a strong opinion. It just felt a bit much to change 300+ files to solve a vulnerability and I would generally prefer this type of change to be as focused as possible. But at the same time, as you said updating might mean additional rules which we would want in the long run anyway. Not blocking either way, it was just an observation.

File: packages/datadog-instrumentations/src/rhea.js [hunk 1]
```diff
@@ -45,7 +45,9 @@ addHook({ name: 'rhea', versions: ['>=1'], file: 'lib/link.js' }, obj => {
     const { host, port } = getHostAndPort(this.connection)
 
     const targetAddress = this.options && this.options.target &&
-      this.options.target.address ? this.options.target.address : undefined
+      this.options.target.address
+      ? this.options.target.address
+      : undefined
```
Comment (simon-id) [inline packages/datadog-instrumentations/src/rhea.js:50]: shouldn't it be 
```suggestion
      this.options.target.address
        ? this.options.target.address
        : undefined
```

File: packages/datadog-plugin-http/src/client.js [hunk 2]
```diff
@@ -227,7 +227,7 @@ function hasAmazonSignature (options) {
       return true
     }
 
-    if ([].concat(headers['authorization']).some(startsWith('AWS4-HMAC-SHA256'))) {
+    if ([].concat(headers.authorization).some(startsWith('AWS4-HMAC-SHA256'))) {
       return true
     }
```
Comment (simon-id) [inline packages/datadog-plugin-http/src/client.js:232]: unrelated to the PR but what the hell is that ?

File: packages/dd-trace/src/appsec/blocked_templates.js
Comment (simon-id) [inline packages/dd-trace/src/appsec/blocked_templates.js:1]: Why this change ? since those templates are copied from an external source, it's quite bothersome to re-escape it everytime it needs to be updated

Comment (Qard) [reply packages/dd-trace/src/appsec/blocked_templates.js:1]: They linter yells at us without those changes. We could maybe configure it to ignore that file though. 🤔

File: packages/datadog-instrumentations/src/rhea.js (hunk 1 above)
Comment (Qard) [reply packages/datadog-instrumentations/src/rhea.js:50]: The linter auto-fixed it like that. Probably just not smart enough to parse the indentation correctly.

File: packages/datadog-plugin-http/src/client.js (hunk 2 above)
Comment (Qard) [reply packages/datadog-plugin-http/src/client.js:232]: Propagating headers with AWS was feature-gated as, if I recall correctly, there was some issue when signatures are present. @rochdev probably remembers better why we are doing this.

**Review by szegedi — APPROVED** [review]
Okay from profiling.

File: packages/dd-trace/src/appsec/blocked_templates.js
Comment (Qard) [reply packages/dd-trace/src/appsec/blocked_templates.js:1]: I've ignored the file.

File: packages/dd-trace/src/config.js [hunk 3]
```diff
@@ -23,13 +23,14 @@ const fromEntries = Object.fromEntries || (entries =>
...
 const defaultWafObfuscatorValueRegex =
-`(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|to\
-ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:\\s*=[^;]|"\\s*:\\s*"[^"]+")|bearer\
-\\s+[a-z0-9\\._\\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\\w=-]+\\.ey[I-L][\\w=-]+(?:\\.[\\w.+\\/=-]+)?\
-|[\\-]{5}BEGIN[a-z\\s]+PRIVATE\\sKEY[\\-]{5}[^\\-]+[\\-]{5}END[a-z\\s]+PRIVATE\\sKEY|ssh-rsa\\s*[a-z0-9\\/\\.+]{100,}`
+'(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|to' +
+'ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:\\s*=[^;]|"\\s*:\\s*"[^"]+")|' +
+'bearer\\s+[a-z0-9\\._\\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\\w=-]+\\.ey[I-L][\\w=-]+(?:\\.[' +
+'\\w.+\\/=-]+)?|[\\-]{5}BEGIN[a-z\\s]+PRIVATE\\sKEY[\\-]{5}[^\\-]+[\\-]{5}END[a-z\\s]+PRIVATE\\sKEY|ssh-rsa\\s*[' +
+'a-z0-9\\/\\.+]{100,}'
```
[the remark this answers is not in the document]
Comment (Qard) [reply packages/dd-trace/src/config.js]: It seemed to already be getting split with \ which is not valid in all JS environments anyway, thus the lint warning. I changed it to single quotes but as a single line with the max-len warning disabled as that seems most copy/paste-friendly. Let me know if that works for you.

**Review by rochdev — APPROVED** [review]

**Review by szegedi — APPROVED** [review]
:ship: from profiling

**Review by juan-fernandez — APPROVED** [review]
looks good from ci visibility's perspective

**Review by duncanista — APPROVED** [review]
For Serverless related code – LGTM

**Review by wconti27 — APPROVED** [review]

File: integration-tests/esbuild/aws-sdk.js [hunk 4]
```diff
@@ -2,4 +2,4 @@ require('../../').init() // dd-trace
 
 const aws = require('aws-sdk')
 
-void aws.util.inherit
+global.test = aws.util.inherit
```
Comment (labbati) [inline integration-tests/esbuild/aws-sdk.js:5]: @Qard can you confirm this is intended

**Review by labbati — APPROVED** [review]

File: integration-tests/esbuild/aws-sdk.js (hunk 4 above)
Comment (Qard) [reply integration-tests/esbuild/aws-sdk.js:5]: Yes. The linter yells about the use of void, but without it esbuild will remove the line as dead code. Assigning it to `global.test` prevents that.

**Review by tonyredondo — APPROVED** [review]

**Review by simon-id — APPROVED** [review]
